### PR TITLE
fix #4460 removing the split packages

### LINKED ISSUE #4460: java.lang.ClassNotFoundException: io.fabric8.kubernetes.client.DefaultKubernetesClient
### Describe the bug

I'm using io.fabric8:kubernetes-client:6.1.1 and am running a simple client creation code:
```
final var kubernetesClient = new KubernetesClientBuilder().build();
System.out.println(kubernetesClient.getNamespace());
```

module-info.java
```
module com.some.module {
    requires kubernetes.client.api;
}
```

Error:
```
java.lang.ClassNotFoundException: io.fabric8.kubernetes.client.DefaultKubernetesClient
```

In module-info.java, if I use `kubernetes.client` instead of `kubernetes.client.api`, then I'mgetting the below error
```
module com.some.module {
    requires kubernetes.client;
}
```
```
java: module xyz reads package io.fabric8.kubernetes.client from both kubernetes.client.api and kubernetes.client
```

Since DefaultKubernetesClient is deprecated in 6.1.1, I'm using KubernetesClientBuilder.

### Fabric8 Kubernetes Client version

6.1.1

### Steps to reproduce

1. Use io.fabric8:kubernetes-client:6.1.1
2. In module-info.java add requires kubernetes.client.api
```
module com.some.module {
    requires kubernetes.client.api;
}
```
3. Try to create a Kubernetes client using KubernetesClientBuilder
```
import io.fabric8.kubernetes.client.KubernetesClientBuilder;

public class Example {
    public static void main(String[] args) {
        final var kubernetesClient = new KubernetesClientBuilder().build();
        System.out.println(kubernetesClient.getNamespace());

    }
}
```

### Expected behavior

Kubernetes client should be created without any errors

### Runtime

other (please specify in additional context)

### Kubernetes API Server version

other (please specify in additional context)

### Environment

Windows

### Fabric8 Kubernetes Client Logs

```shell
Exception in thread "main" io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
	at kubernetes.client.api@6.1.1/io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:129)
	at kubernetes.client.api@6.1.1/io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:122)
	at kubernetes.client.api@6.1.1/io.fabric8.kubernetes.client.KubernetesClientBuilder.<init>(KubernetesClientBuilder.java:58)
	at com.some.module/com.some.module.Main.main(Main.java:106)
Caused by: java.lang.ClassNotFoundException: io.fabric8.kubernetes.client.DefaultKubernetesClient
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	at kubernetes.client.api@6.1.1/io.fabric8.kubernetes.client.KubernetesClientBuilder.<init>(KubernetesClientBuilder.java:56)
	... 1 more
```

### Additional context

```
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.0", GitCommit:"cb303e613a121a29364f75cc67d3d580833a7479", GitTreeState:"clean", BuildDate:"2021-04-08T16:31:21Z", GoVersion:"go1.16.1", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.1", GitCommit:"5e58841cce77d4bc13713ad2b91fa0d961e69192", GitTreeState:"clean", BuildDate:"2021-05-12T14:12:29Z", GoVersion:"go1.16.4", Compiler:"gc", Platform:"linux/amd64"}
```
```
<dependency>
      <groupId>io.fabric8</groupId>
      <artifactId>knative-client</artifactId>
      <version>6.1.1</version>
</dependency>
```

### PR DESCRIPTION
**Pull request by shawkins**
## Description

Fix #4460

To support java modules (#4460) and cleaner OSGI support split packages need to be removed.  Since we've now had 6.0 / 6.1 releases with the relevant client classes either explicitly or effectively deprecated we should be able to resolve this for splits that were introduced by the client/api refactorings.

kubernetes-client:
- io.fabric8.kubernetes.client -> io.fabric8.kubernetes.client.impl
- io.fabric8.kubernetes.client.internal -> io.fabric8.kubernetes.client.dsl.internal (only 3 classes)

openshift-client:
- io.fabric8.openshift.client -> io.fabric8.openshift.client.impl

Additionally a still deprecated DefaultKubernetesClient and DefaultOpenShiftClient adapters were added to the -api modules for continued compatibility when not using a builder.  The actual implementations have been renamed XXXClientImpl.

I tested against a sample module project and this seemed to be sufficient to address the issue.  There may be some more additional changes needed for OSGI - in particular the osgi.export and osgi.private should be reviewed for kubernetes-client and openshift-client @rohanKanojia is this something that you are familiar with?

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [ ] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](https://github.com/fabric8io/kubernetes-client/blob/master/CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: kubernetes-client/src/test/java/io/fabric8/kubernetes/client/impl/DefaultKubernetesClientTest.java [hunk 1]
```diff
@@ -13,11 +13,14 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.fabric8.kubernetes.client;
+package io.fabric8.kubernetes.client.impl;
```
Comment (manusa) [inline kubernetes-client/src/test/java/io/fabric8/kubernetes/client/impl/DefaultKubernetesClientTest.java:16]: The test class name should probably be renamed

**Review by manusa — APPROVED** [review]
LGTM

I'd prefer to see something like `VanillaKubernetesClient` instead of `KubernetesClientImpl`, but that's something very personal (I hate the `Impl` suffix in class names).

File: kubernetes-client/src/test/java/io/fabric8/kubernetes/client/impl/DefaultKubernetesClientTest.java (hunk 1 above)
Comment (shawkins) [reply kubernetes-client/src/test/java/io/fabric8/kubernetes/client/impl/DefaultKubernetesClientTest.java]: Done

**Comment by shawkins** [comment]
> I'd prefer to see something like VanillaKubernetesClient instead of KubernetesClientImpl, but that's something very personal (I hate the Impl suffix in class names).

Feel free to use whatever naming you want - I don't have a strong preference.  We'll probably want to re-assess the package names as well at some point.  We could just have io.fabric8.kubernetes.client.(impl or internal) as the root package, rather than the two impls and two internals that will be there after this pr.

**Pull request merged**